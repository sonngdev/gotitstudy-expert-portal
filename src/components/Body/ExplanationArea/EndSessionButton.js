import React, { useState } from 'react';
import { Button, Modal } from '@gotitinc/design-system';

export function EndSessionButton() {
  const [modalVisible, setModalVisible] = useState(false);
  const showModal = () => setModalVisible(true);
  const hideModal = () => setModalVisible(false);
  const endSession = () => window.location.reload();

  return (
    <>
      <Button className="end-session" onClick={showModal}>End session</Button>

      <Modal size="small" show={modalVisible}>
        <Modal.Header closeButton>
          <Modal.Title>Stop working</Modal.Title>
        </Modal.Header>

        <Modal.Body>
          Are you sure you want to end this session? This action cannot be undone.
        </Modal.Body>

        <Modal.Footer className="u-flex">
          <Button variant="secondary" className="cancel" onClick={hideModal}>
            Cancel
          </Button>
          <Button variant="primary" className="confirm" onClick={endSession}>
            Stop working
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  );
}
