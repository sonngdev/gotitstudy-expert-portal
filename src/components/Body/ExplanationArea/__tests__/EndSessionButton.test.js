import React from 'react';
import { shallow } from 'enzyme';
import { Modal } from '@gotitinc/design-system';
import { EndSessionButton } from '../EndSessionButton';

describe('<EndSessionButton />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<EndSessionButton />);
  });

  it('renders', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('contains a button', () => {
    const button = wrapper.find('.end-session');
    expect(button.exists()).toBe(true);
    expect(button.text()).toBe('End session');
  });

  it('has a modal that is initially hidden', () => {
    const modal = wrapper.find('Modal');
    expect(modal.exists()).toBe(true);
    expect(modal.prop('show')).toBe(false);
  });

  describe('<Modal />', () => {
    beforeEach(() => {
      wrapper.find('.end-session').simulate('click');
    });

    it('shows up when button is clicked', () => {
      expect(wrapper.find('Modal').prop('show')).toBe(true);
    });

    it('contains a title', () => {
      const title = wrapper.find(Modal.Title);
      expect(title.exists()).toBe(true);
      expect(title.text()).toBe('Stop working');
    });

    it('contains a body message', () => {
      const body = wrapper.find(Modal.Body);
      expect(body.exists()).toBe(true);
      expect(body.text()).toBe('Are you sure you want to end this session? This action cannot be undone.');
    });

    it('hides when cancel button is clicked', () => {
      wrapper.find('Modal').find('.cancel').simulate('click');
      expect(wrapper.find('Modal').prop('show')).toBe(false);
    });

    it('reloads the page when confirm button is clicked', () => {
      const { location } = window;
      delete window.location;
      window.location = { reload: jest.fn() };

      const confirm = wrapper.find('Modal').find('.confirm');
      confirm.simulate('click');
      expect(window.location.reload).toHaveBeenCalled();

      window.location = location;
    });
  });
});
