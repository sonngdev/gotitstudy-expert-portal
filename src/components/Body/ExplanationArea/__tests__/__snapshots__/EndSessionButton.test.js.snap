// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EndSessionButton /> renders 1`] = `
"<Fragment>
  <Button className=\\"end-session\\" onClick={[Function: showModal]} nonUppercase={false} variant=\\"primary\\">
    End session
  </Button>
  <Modal size=\\"small\\" show={false} relative={false} transition={{...}}>
    <Modal.Header closeButton={true}>
      <ModalTitle uText650 uTextCenter>
        Stop working
      </ModalTitle uText650 uTextCenter>
    </Modal.Header>
    <ModalBody uPaddingHorizontalMedium uBackgroundWhite uPaddingTopSmall uPaddingBottomMedium>
      Are you sure you want to end this session? This action cannot be undone.
    </ModalBody uPaddingHorizontalMedium uBackgroundWhite uPaddingTopSmall uPaddingBottomMedium>
    <ModalFooter uBackgroundLightest uPaddingMedium uFlex uAlignItemsCenter uJustifyContentEnd className=\\"u-flex\\">
      <Button variant=\\"secondary\\" className=\\"cancel\\" onClick={[Function: hideModal]} nonUppercase={false}>
        Cancel
      </Button>
      <Button variant=\\"primary\\" className=\\"confirm\\" onClick={[Function: endSession]} nonUppercase={false}>
        Stop working
      </Button>
    </ModalFooter uBackgroundLightest uPaddingMedium uFlex uAlignItemsCenter uJustifyContentEnd>
  </Modal>
</Fragment>"
`;
